[16.0][REF] l10n_br_account, l10n_br_account_nfe: relax the l10n_br_account_due_list dependency

rvalyi opened the pull request:
No intuito de deixar os módulos centrais da localização mais fáceis de migrar e menos invasivos:

- tirei a dependência pro l10n_br_account_due_list do l10n_br_account. Realmente não ta sendo usado pelo modulo hoje e é uma boa coisa a esse nível
-  movei a dependência ao l10n_br_account_due_list no l10n_br_account_nfe onde realmente é usado para mapear as tags cobr e pag.

Além disso é bom observar que o modulo l10n_br_account_payment_order depende do l10n_br_account_due_list e que a dependência já estava listada (não tive que mudar nada aqui).

Finalmente vale a pena mencionar esse meu outro PR onde eu limpei o modulo l10n_br_account_due_list: #3843

rvalyi committed:
[REF] l10n_br_account: rm l10n_br_account_due_list dep here

rvalyi committed:
[REF] l10n_br_account_nfe: add l10n_br_account_due_list dep

antoniospneto reviewed (changes requested):
ops, acusou erro nos testes

rvalyi committed:
TEST

wip

rvalyi commented:
O porque a dependência do l10n_br_account pro l10n_br_account_due_list é necessária para não ter unbalanced moves ainda é um grande mistério... Eu tentei:

- depender apenas do account_due_list no l10n_br_account e deu os mesmos erros
- depender do l10n_br_account_due_list mas comentar a visão e o calculo do payment_move_line_ids e os testes passaram. Ou seja não sei que droga no l10n_br_account_due_list faz os testes passar...
- se instalar apenas o l10n_br_account sem os outros módulos,  sem a dependência do account_due_list os testes passam certinho...

cc @antoniospneto

rvalyi committed:
WIP TEST

rvalyi committed:
[FIX] l10n_br_stock: Command.link fix

rvalyi committed:
WIP TEST -> OK!

rvalyi committed:
TEST WIP -> OK!

rvalyi committed:
WIP TEST -> OK!

rvalyi committed:
WIP TEST -> KO!

 2025-06-29 22:23:47,679 572 INFO odoo odoo.addons.l10n_br_account.tests.test_invoice_refund: ======================================================================
2025-06-29 22:23:47,679 572 ERROR odoo odoo.addons.l10n_br_account.tests.test_invoice_refund: ERROR: TestInvoiceRefund.test_refund
Traceback (most recent call last):
  File "/opt/odoo-venv/lib/python3.10/site-packages/odoo/addons/l10n_br_account/tests/test_invoice_refund.py", line 112, in test_refund
    invoice.fiscal_operation_id = self.env.ref("l10n_br_fiscal.fo_venda")
  File "/opt/odoo/odoo/fields.py", line 1337, in __set__
    records.write({self.name: write_value})
  File "/opt/odoo/addons/purchase/models/account_invoice.py", line 149, in write
    res = super(AccountMove, self).write(vals)
  File "/opt/odoo/addons/account/models/account_move.py", line 2359, in write
    with self.env.protecting(self._get_protected_vals(vals, self)), self._check_balanced(container):
  File "/usr/lib/python3.10/contextlib.py", line 142, in __exit__
    next(self.gen)
  File "/opt/odoo/addons/account/models/account_move.py", line 1773, in _check_balanced
    raise UserError(error_msg)
odoo.exceptions.UserError: An error has occurred.

The move (Test Refund Invoice) is not balanced.
The total of debits equals R$ 0.00 and the total of credits equals R$ 15.00.
You might want to specify a default account on journal "Refund Journal - (test)" to automatically balance each move.

2025-06-29 22:23:47,683 572 INFO odoo odoo.addons.l10n_br_account.tests.test_invoice_refund: Starting TestInvoiceRefund.test_refund_force_fiscal_operation ...
2025-06-29 22:23:47,753 572 INFO odoo odoo.addons.l10n_br_account.tests.test_invoice_refund: ======================================================================
2025-06-29 22:23:47,753 572 ERROR odoo odoo.addons.l10n_br_account.tests.test_invoice_refund: ERROR: TestInvoiceRefund.test_refund_force_fiscal_operation
Traceback (most recent call last):
  File "/opt/odoo-venv/lib/python3.10/site-packages/odoo/addons/l10n_br_account/tests/test_invoice_refund.py", line 141, in test_refund_force_fiscal_operation
    invoice.fiscal_operation_id = self.env.ref("l10n_br_fiscal.fo_venda")
  File "/opt/odoo/odoo/fields.py", line 1337, in __set__
    records.write({self.name: write_value})
  File "/opt/odoo/addons/purchase/models/account_invoice.py", line 149, in write
    res = super(AccountMove, self).write(vals)
  File "/opt/odoo/addons/account/models/account_move.py", line 2359, in write
    with self.env.protecting(self._get_protected_vals(vals, self)), self._check_balanced(container):
  File "/usr/lib/python3.10/contextlib.py", line 142, in __exit__
    next(self.gen)
  File "/opt/odoo/addons/account/models/account_move.py", line 1773, in _check_balanced
    raise UserError(error_msg)
odoo.exceptions.UserError: An error has occurred.

The move (Draft Invoice Test Refund Invoice) is not balanced.
The total of debits equals R$ 0.00 and the total of credits equals R$ 15.00.
You might want to specify a default account on journal "Refund Journal - (test)" to automatically balance each move.

rvalyi committed:
WIP TEST -> KO!

 2025-06-29 22:43:22,958 630 INFO odoo odoo.addons.l10n_br_account.tests.test_invoice_refund: Starting TestInvoiceRefund.test_refund ...
2025-06-29 22:43:23,361 630 INFO odoo odoo.models.unlink: User #1 deleted account.move.line records with IDs: [58]
2025-06-29 22:43:23,493 630 INFO odoo odoo.addons.l10n_br_account.tests.test_invoice_refund: ======================================================================
2025-06-29 22:43:23,493 630 ERROR odoo odoo.addons.l10n_br_account.tests.test_invoice_refund: ERROR: TestInvoiceRefund.test_refund
Traceback (most recent call last):
  File "/opt/odoo-venv/lib/python3.10/site-packages/odoo/addons/l10n_br_account/tests/test_invoice_refund.py", line 112, in test_refund
    invoice.fiscal_operation_id = self.env.ref("l10n_br_fiscal.fo_venda")
  File "/opt/odoo/odoo/fields.py", line 1337, in __set__
    records.write({self.name: write_value})
  File "/opt/odoo/addons/purchase/models/account_invoice.py", line 149, in write
    res = super(AccountMove, self).write(vals)
  File "/opt/odoo/addons/account/models/account_move.py", line 2359, in write
    with self.env.protecting(self._get_protected_vals(vals, self)), self._check_balanced(container):
  File "/usr/lib/python3.10/contextlib.py", line 142, in __exit__
    next(self.gen)
  File "/opt/odoo/addons/account/models/account_move.py", line 1773, in _check_balanced
    raise UserError(error_msg)
odoo.exceptions.UserError: An error has occurred.

The move (Test Refund Invoice) is not balanced.
The total of debits equals R$ 0.00 and the total of credits equals R$ 15.00.
You might want to specify a default account on journal "Refund Journal - (test)" to automatically balance each move.

2025-06-29 22:43:23,497 630 INFO odoo odoo.addons.l10n_br_account.tests.test_invoice_refund: Starting TestInvoiceRefund.test_refund_force_fiscal_operation ...
2025-06-29 22:43:23,574 630 INFO odoo odoo.addons.l10n_br_account.tests.test_invoice_refund: ======================================================================
2025-06-29 22:43:23,574 630 ERROR odoo odoo.addons.l10n_br_account.tests.test_invoice_refund: ERROR: TestInvoiceRefund.test_refund_force_fiscal_operation
Traceback (most recent call last):
  File "/opt/odoo-venv/lib/python3.10/site-packages/odoo/addons/l10n_br_account/tests/test_invoice_refund.py", line 141, in test_refund_force_fiscal_operation
    invoice.fiscal_operation_id = self.env.ref("l10n_br_fiscal.fo_venda")
  File "/opt/odoo/odoo/fields.py", line 1337, in __set__
    records.write({self.name: write_value})
  File "/opt/odoo/addons/purchase/models/account_invoice.py", line 149, in write
    res = super(AccountMove, self).write(vals)
  File "/opt/odoo/addons/account/models/account_move.py", line 2359, in write
    with self.env.protecting(self._get_protected_vals(vals, self)), self._check_balanced(container):
  File "/usr/lib/python3.10/contextlib.py", line 142, in __exit__
    next(self.gen)
  File "/opt/odoo/addons/account/models/account_move.py", line 1773, in _check_balanced
    raise UserError(error_msg)
odoo.exceptions.UserError: An error has occurred.

The move (Draft Invoice Test Refund Invoice) is not balanced.
The total of debits equals R$ 0.00 and the total of credits equals R$ 15.00.
You might want to specify a default account on journal "Refund Journal - (test)" to automatically balance each move.

rvalyi committed:
TEST WIP  -> KO!

 2025-06-29 23:02:36,976 650 ERROR odoo odoo.addons.l10n_br_account.tests.test_invoice_refund: ERROR: TestInvoiceRefund.test_refund
Traceback (most recent call last):
  File "/opt/odoo-venv/lib/python3.10/site-packages/odoo/addons/l10n_br_account/tests/test_invoice_refund.py", line 112, in test_refund
    invoice.fiscal_operation_id = self.env.ref("l10n_br_fiscal.fo_venda")
  File "/opt/odoo/odoo/fields.py", line 1337, in __set__
    records.write({self.name: write_value})
  File "/opt/odoo/addons/purchase/models/account_invoice.py", line 149, in write
    res = super(AccountMove, self).write(vals)
  File "/opt/odoo/addons/account/models/account_move.py", line 2359, in write
    with self.env.protecting(self._get_protected_vals(vals, self)), self._check_balanced(container):
  File "/usr/lib/python3.10/contextlib.py", line 142, in __exit__
    next(self.gen)
  File "/opt/odoo/addons/account/models/account_move.py", line 1773, in _check_balanced
    raise UserError(error_msg)
odoo.exceptions.UserError: An error has occurred.

The move (Test Refund Invoice) is not balanced.
The total of debits equals R$ 0.00 and the total of credits equals R$ 15.00.
You might want to specify a default account on journal "Refund Journal - (test)" to automatically balance each move.

2025-06-29 23:02:36,979 650 INFO odoo odoo.addons.l10n_br_account.tests.test_invoice_refund: Starting TestInvoiceRefund.test_refund_force_fiscal_operation ...
2025-06-29 23:02:37,046 650 INFO odoo odoo.addons.l10n_br_account.tests.test_invoice_refund: ======================================================================
2025-06-29 23:02:37,047 650 ERROR odoo odoo.addons.l10n_br_account.tests.test_invoice_refund: ERROR: TestInvoiceRefund.test_refund_force_fiscal_operation
Traceback (most recent call last):
  File "/opt/odoo-venv/lib/python3.10/site-packages/odoo/addons/l10n_br_account/tests/test_invoice_refund.py", line 141, in test_refund_force_fiscal_operation
    invoice.fiscal_operation_id = self.env.ref("l10n_br_fiscal.fo_venda")
  File "/opt/odoo/odoo/fields.py", line 1337, in __set__
    records.write({self.name: write_value})
  File "/opt/odoo/addons/purchase/models/account_invoice.py", line 149, in write
    res = super(AccountMove, self).write(vals)
  File "/opt/odoo/addons/account/models/account_move.py", line 2359, in write
    with self.env.protecting(self._get_protected_vals(vals, self)), self._check_balanced(container):
  File "/usr/lib/python3.10/contextlib.py", line 142, in __exit__
    next(self.gen)
  File "/opt/odoo/addons/account/models/account_move.py", line 1773, in _check_balanced
    raise UserError(error_msg)
odoo.exceptions.UserError: An error has occurred.

The move (Draft Invoice Test Refund Invoice) is not balanced.
The total of debits equals R$ 0.00 and the total of credits equals R$ 15.00.
You might want to specify a default account on journal "Refund Journal - (test)" to automatically balance each move.

rvalyi committed:
WIP TEST -> MEHHH

 2025-06-29 23:21:05,816 634 INFO odoo odoo.modules.loading: Loading module l10n_br_sped_base (55/127)
2025-06-29 23:21:06,329 634 INFO odoo odoo.modules.registry: module l10n_br_sped_base: creating or updating database tables
2025-06-29 23:21:06,561 634 INFO odoo odoo.modules.loading: loading l10n_br_sped_base/views/sped_base.xml
2025-06-29 23:21:06,581 634 INFO odoo odoo.tests.suite: ======================================================================
2025-06-29 23:21:06,581 634 ERROR odoo odoo.tests.suite: ERROR: setUpClass (odoo.addons.l10n_br_sped_base.tests.test_sped_base.TestSpedBase)
Traceback (most recent call last):
  File "/__w/l10n-brazil/l10n-brazil/l10n_br_sped_base/tests/test_sped_base.py", line 26, in setUpClass
    cls.loader.backup_registry()
  File "/opt/odoo-venv/lib/python3.10/site-packages/odoo_test_helper/fake_model_loader.py", line 99, in backup_registry
    self._check_wrong_import()
  File "/opt/odoo-venv/lib/python3.10/site-packages/odoo_test_helper/fake_model_loader.py", line 91, in _check_wrong_import
    if module_path not in modules_to_ignore and path[3] == "tests":
IndexError: list index out of range

rvalyi committed:
WIP TEST: disable sped modules even if looks unrelated  -> OK!

rvalyi committed:
WIP test

rvalyi commented:
eu fui limitando os modulos instalaveis para ver onde que a remoção dava problema. Comentei (amendei) nos commits quando a CI passou ou não. E me parece que:

- se remover a dependência no l10n_br_account, o l10n_br_account_nfe da um problema ao instalar.
- se instalar o l10n_br_nfe antes, parece que cria um problema de account.move não equlibrados nos testes do l10n_br_account. Talvez é o combo de l10n_br_nfe com algum outro modulo.
- se instalar o l10n_br_cte sem o l10n_br_nfe tem um bug que não esta relacionado mas que eu descobri com isso [0]

[0] https://github.com/OCA/l10n-brazil/actions/runs/15961616097/job/45014989084?pr=3845#step:9:1424 
```
 ERROR: insert or update on table "l10n_br_fiscal_document" violates foreign key constraint "l10n_br_fiscal_document_document_type_id_fkey"
DETAIL:  Key (document_type_id)=(43) is not present in table "l10n_br_fiscal_document_type".
```

rvalyi commented:
isso acabou entrando no #4115,  podemos fechar esse então.
